akka-javasdk/src/main/java/akka/javasdk/client/ComponentClient.java

```diff
@@ -58,4 +60,12 @@ public interface ComponentClient {
 
   /** Select {@link akka.javasdk.view.View} as a call target component. */
   ViewClient forView();
+
+  /**
+   * Select {@link akka.javasdk.agent.ChatAgent} as a call target component.
+   *
+   * @param sessionId - The agent may participate in a session, which is used for the agent's
+   *     conversational memory.
+   */
+  ChatAgentClient forChatAgent(Optional<String> sessionId);
```

Comment:
One 0 and one 1 parameter overload would be better ergonomics

Reply:
I was thinking that sessionId is important to notice since that will drive if we use memory or not, and therefore I made it an explicit choice. If we agree that is useful, we could still provide overloads with `Option<String>` and `String`.

Or we simply use null and empty string as the opt out from session?